SparseGrids/Benchmarks/bench_main.cpp

```diff
@@ -24,14 +25,26 @@ int main(int argc, const char** argv){
     }
 
     bool pass = true; // check if the rest of the inputs are OK
-    if (test == bench_make)
-        pass = benchmark_makegrid(args);
-    if (test == bench_loadneeded)
-        pass = benchmark_loadneeded(args);
-    if (test == bench_evaluate || test == bench_evaluate_mixed)
-        pass = benchmark_evaluate(args, (test == bench_evaluate_mixed));
-    if (test == bench_iweights)
-        pass = benchmark_iweights(args);
+    switch(test){
+        case bench_make:
+            pass = benchmark_makegrid(args);
+            break;
+        case bench_loadneeded:
+            pass = benchmark_loadneeded(args);
+            break;
+        case bench_evaluate:
+        case bench_evaluate_mixed:
+            pass = benchmark_evaluate(args, (test == bench_evaluate_mixed));
+            break;
+        case bench_differentiate:
+            pass = benchmark_differentiate(args);
+            break;
+        case bench_iweights:
+            pass = benchmark_iweights(args);
+            break;
+        default:
+            throw std::runtime_error("bench_main.cpp: invalid test type is switch statement!");
```

Comment:
Should the input be `"bench_main.cpp: invalid test type in switch statement!"`?

Reply:
Yes!